File: 238-product-of-array-except-self/238-product-of-array-except-self.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
const productExceptSelf = function(nums) {
    const result = Array(nums.length).fill(1);
    let prefix = 1;
    for (let i = 0; i < nums.length; i++) {
        result[i] = prefix;
        prefix *= nums[i];
    }
    let postfix = 1;
    for (let i = nums.length - 1; i >= 0; i--) {
        result[i] *= postfix;
        postfix *= nums[i];
    }
    return result;
};

File: 238-product-of-array-except-self/238-product-of-array-except-self.rb
# @param {Integer[]} nums
# @return {Integer[]}
def product_except_self(nums)
    n = nums.size
    res = [1] * n
    l = r = 1
    1.upto(n - 1) do |i|
        l *= nums[i - 1]
        res[i] *= l
        # '~' wraps the index around
        r *= nums[-i]
        res[~i] *= r
    end
    res
end